src/webgl/material.js

```diff
@@ -718,171 +725,521 @@ p5.prototype.createFilterShader = function (fragSrc) {
- * uniform mat4 uProjectionMatrix;
- *
  * attribute vec3 aPosition;
- * attribute vec2 aTexCoord;
- * varying vec2 vTexCoord;
- *
  * void main() {
- *   vTexCoord = aTexCoord;
- *   vec4 positionVec4 = vec4(aPosition, 1.0);
- *   gl_Position = uProjectionMatrix * uModelViewMatrix * positionVec4;
+ *   gl_Position = vec4(aPosition, 1.0);
  * }
  * `;
  *
- * // Create a string with the fragment shader program.
- * // The fragment shader is called for each pixel.
  * let fragSrc = `
  * precision highp float;
- *
+ * uniform float uMouseX;
  * void main() {
- *   // Set each pixel's RGBA value to yellow.
- *   gl_FragColor = vec4(1.0, 1.0, 0.0, 1.0);
+ *   vec3 color = vec3(uMouseX, 0.0, 1.0); // Color based on mouse X
+ *   gl_FragColor = vec4(color, 1.0);
  * }
  * `;
  *
  * function setup() {
  *   createCanvas(100, 100, WEBGL);
+ *   fillShader = createShader(vertSrc, fragSrc);
+ *   shader(fillShader);
+ *   noStroke();
+ *   describe('A square with color changing based on mouse X without lighting.');
+ * }
  *
- *   // Create a p5.Shader object.
- *   let shaderProgram = createShader(vertSrc, fragSrc);
+ * function draw() {
+ *   fillShader.setUniform('uMouseX', map(mouseX, 0, width, 0, 1));
+ *   plane(100, 100);
+ * }
+ * </code>
+ * </div>
  *
- *   // Apply the p5.Shader object.
- *   shader(shaderProgram);
+ * @example
+ * <div modernizr='webgl'>
+ * <code>
+ * // Example 2: Fill Shader with Lights Affected by Mouse
+ * // Demonstrating interaction between light and shader, with the light position controlled by the mouse
  *
- *   // Style the drawing surface.
+ * let vertSrc = `
+ * precision highp float;
+ * attribute vec3 aPosition;
+ * uniform mat4 uModelViewMatrix;
+ * uniform mat4 uProjectionMatrix;
+ * varying vec3 vPosition;
+ *
+ * void main() {
+ *   vPosition = aPosition;
+ *   gl_Position = uProjectionMatrix * uModelViewMatrix * vec4(aPosition, 1.0);
+ * }
+ * `;
+ *
+ * let fragSrc = `
+ * precision highp float;
+ * uniform vec3 uLightPos;
+ * varying vec3 vPosition;
+ * void main() {
+ *   float brightness = dot(normalize(uLightPos), normalize(vPosition));
+ *   brightness = clamp(brightness, 0.0, 1.0);
+ *   vec3 color = vec3(brightness, brightness * 0.7, 1.0); // Blue-based color affected by light
+ *   gl_FragColor = vec4(color, 1.0);
+ * }
+ * `;
+ *
+ * let fillShader;
+ *
+ * function setup() {
+ *   createCanvas(100, 100, WEBGL);
+ *   fillShader = createShader(vertSrc, fragSrc);
+ *   shader(fillShader);
  *   noStroke();
+ *   describe('A square with fill color affected by light, light position changes with mouse.');
+ * }
  *
- *   // Add a plane as a drawing surface.
+ * function draw() {
+ *   let lightPos = [(mouseX - width / 2) / width, (mouseY - height / 2) / height, 1.0];
+ *   fillShader.setUniform('uLightPos', lightPos); // Mouse changes light position
  *   plane(100, 100);
- *
- *   describe('A yellow square.');
  * }
  * </code>
  * </div>
  *
- * <div>
+ * @example
+ * <div modernizr='webgl'>
```

Comment:
For this one, from my idea, I think you can draw a sphere(with strokes) and with a panorama(img) and you can show that the image on the panorama is not effected, also the strokes of the sphere is also not effected by shader. The only thing effected by `shader()` is the color of the sphere. Let me know what you think.